srcStatic/GameModules/DllModule.cpp

```diff
@@ -10,7 +11,8 @@ DllModule::DllModule(const std::string& moduleName)
 void DllModule::LoadModuleDll(const std::string& dllPath)
 {
 	// Try to load a DLL with the given name (possibly an empty string)
-	moduleDllHandle.reset(LoadLibraryA(dllPath.c_str()));
+	moduleDllHandle.reset(
+		LoadLibraryExA(dllPath.c_str(), NULL, fs::path(dllPath).is_absolute() ? LOAD_WITH_ALTERED_SEARCH_PATH : 0));
```

Comment:
You're misunderstanding. In practice, `dllPath` will always be absolute, but I added a check there to be totally safe, since `LOAD_WITH_ALTERED_SEARCH_PATH` results in UB if the provided path is not absolute.

Reply:
Reference documentation:
https://docs.microsoft.com/en-us/windows/win32/api/libloaderapi/nf-libloaderapi-loadlibraryexa

> LOAD_WITH_ALTERED_SEARCH_PATH
>
> If this value is used and lpFileName specifies an absolute path, the system uses the alternate file search strategy discussed in the Remarks section to find associated executable modules that the specified module causes to be loaded. If this value is used and lpFileName specifies a relative path, the behavior is undefined.
If this value is not used, or if lpFileName does not specify a path, the system uses the standard search strategy discussed in the Remarks section to find associated executable modules that the specified module causes to be loaded.
>
> This value cannot be combined with any LOAD_LIBRARY_SEARCH flag.

----

For library search order:
https://docs.microsoft.com/en-ca/windows/win32/dlls/dynamic-link-library-search-order?redirectedfrom=MSDN

----

It's not super clear why `LOAD_WITH_ALTERED_SEARCH_PATH` is being used here.